event_handling/src/Timer.cpp

```diff
@@ -1,97 +1,94 @@
 #include "essentials/Timer.h"
 
+#include <thread>
+
```

Comment:
looks like same as NotifyTimer, NotifyTimer can use this one as it is.

Reply:
Okay I made NotifyTimer derive from ITrigger and added another constructor.